galley/pkg/server/components/validation.go

```diff
@@ -15,36 +15,167 @@
 package components
 
 import (
-	"k8s.io/client-go/kubernetes"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
 
-	"istio.io/istio/galley/pkg/crd/validation"
 	"istio.io/istio/galley/pkg/server/process"
+	"istio.io/istio/mixer/pkg/validate"
 	"istio.io/istio/pkg/cmd"
+	"istio.io/istio/pkg/config/schemas"
+	"istio.io/istio/pkg/webhooks/validation/controller"
+	"istio.io/istio/pkg/webhooks/validation/server"
+
+	"istio.io/pkg/log"
 	"istio.io/pkg/probe"
 )
 
-// NewValidation returns a new validation component.
-func NewValidation(kubeInterface kubernetes.Interface, kubeConfig string,
-	params *validation.WebhookParameters, liveness, readiness probe.Controller) process.Component {
+// This is for lint fix
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+func monitorReadiness(stop <-chan struct{}, port uint, readiness *probe.Probe) {
+	const httpsHandlerReadinessFreq = time.Second
+
+	ready := false
+	client := &http.Client{
+		Timeout: time.Second,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+
+	for {
+		if err := webhookHTTPSHandlerReady(client, port); err != nil {
+			readiness.SetAvailable(errors.New("not ready"))
+			scope.Infof("https handler for validation webhook is not ready: %v\n", err)
+			ready = false
+		} else {
+			readiness.SetAvailable(nil)
+			if !ready {
+				scope.Info("https handler for validation webhook is ready\n")
+				ready = true
+			}
+		}
+
+		select {
+		case <-stop:
+			return
+		case <-time.After(httpsHandlerReadinessFreq):
+			// check again
+		}
+	}
+}
+
+func webhookHTTPSHandlerReady(client httpClient, port uint) error {
+	readinessURL := &url.URL{
+		Scheme: "https",
+		Host:   fmt.Sprintf("localhost:%v", port),
+		Path:   server.HTTPSHandlerReadyPath,
+	}
+
+	req := &http.Request{
+		Method: http.MethodGet,
+		URL:    readinessURL,
+	}
+
+	response, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("HTTP request to %v failed: %v", readinessURL, err)
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		return fmt.Errorf("GET %v returned non-200 status=%v",
+			readinessURL, response.StatusCode)
+	}
+	return nil
+}
+
+func NewValidationServer(
+	options server.Options,
+	liveness probe.Controller,
+	readiness probe.Controller,
+) process.Component {
+	return process.ComponentFromFns(
+		// start
+		func() error {
+			stop := make(chan struct{})
+
+			log.Infof("Galley validation server started with \n%s", &options)
+
+			options.MixerValidator = validate.NewDefaultValidator(false)
+			options.PilotDescriptor = schemas.Istio
```

Comment:
Can you use the galley schemas here (`collections.Istio`)? Selfishly trying to avoid more work for my merge :)